### PR TITLE
Added optimizations and examples to string_starts/ends_with and char_is_uppercase.

### PR DESCRIPTION
**Pull request by Lildirt**
Follows PR #420 and #422

### CODE REVIEW

File: src/main/java/com/laytonsmith/core/functions/StringHandling.java [hunk 1]
```diff
@@ -867,13 +867,26 @@ public String docs() {
                 
                 @Override
                 public Construct exec(Target t, Environment env, Construct... args) throws CancelCommandException, ConfigRuntimeException {
+			if (!(args[0] instanceof CString)) {
+				throw new CREFormatException(this.getName() + " expects a string as first argument, but type "
```
Comment (LadyCailin) [inline src/main/java/com/laytonsmith/core/functions/StringHandling.java]: CastException, not format.

File: src/main/java/com/laytonsmith/core/functions/StringHandling.java [hunk 2]
```diff
@@ -917,13 +938,26 @@ public String docs() {
...
```
Comment (LadyCailin) [inline src/main/java/com/laytonsmith/core/functions/StringHandling.java]: CastException

File: src/test/java/com/laytonsmith/core/functions/StringHandlingTest.java [hunk 3]
```diff
@@ -298,6 +299,61 @@ public void testCharIsUppercase() throws Exception {
...
+        @Test
+        public void testStringStartsWith() throws Exception {
+                assertEquals("true", SRun("string_starts_with('magical string here', 'magical')", null));
+                assertEquals("false", SRun("string_starts_with('something', 'pg-13')", null));
+                
+                assertEquals("false", SRun("string_starts_with('magic', true)", null));
+                assertEquals("true", SRun("string_starts_with('true', true)", null));
+                assertEquals("true", SRun("string_starts_with('123', 1)", null));
+                
+                assertEquals("true", SRun("string_starts_with('music', '')", null));
```
Comment (LadyCailin) [inline src/test/java/com/laytonsmith/core/functions/StringHandlingTest.java]: We should probably document this behavior in the function docs

**Comment by cailin-telenor** [comment]
@Lildirt Can you fix conflicts, then I will merge this in?

**Pull request merged**